File: app/src/main/java/com/pranav/java/ide/ExecuteActivity.java
package com.pranav.java.ide;

import androidx.appcompat.app.AppCompatActivity;

import io.github.rosemoe.sora.widget.CodeEditor;

import java.io.Console;

public class ExecuteActivity extends AppCompatActivity {

    private CodeEditor terminal;
    private String logs;

    @Override
    public void onCreate(Bundle savedInstanceState) {
      super.onCreate(savedInstanceState);
      
      terminal = findViewById(R.id.terminal);
      setContentView(R.layout.activity_execute);
      
      Console console = System.console();
      logs = "";
    }
    
    public void write(String text) {
      logs.concat(text);
      terminal.append(text);
    }
}

File: app/src/main/java/com/pranav/java/ide/ExecuteActivity.kt
package com.pranav.java.ide

import androidx.appcompat.app.AppCompatActivity

import io.github.rosemoe.sora.widget.CodeEditor

import java.io.Console

class ExecuteActivity: AppCompatActivity() {

    private val terminal: CodeEditor
    private var logs: String

    override fun onCreate(savedInstanceState: Bundle) {
      super.onCreate(savedInstanceState)
      
      terminal = findViewById(R.id.terminal)
      setContentView(R.layout.activity_execute)
      
      val console = System.console()
      logs = ""
    }
    
    public fun write(text: String) {
      logs.concat(text)
      terminal.append(text)
    }
}